### ISSUE TITLE
why batch_size has to be 1 when update fisher?

### ISSUE THREAD

**Issue by yuchenlin** [GMvandeVen/continual-learning#16]
Hi, 

Thanks for the great repo. I have a quick question about the computation in the Fisher Information Matrix update: does the batch_size have to be 1 for the dataloader here: https://github.com/GMvandeVen/continual-learning/blob/d281967802396b146b2c30b6667369c6f2395472/continual_learner.py#L89 ? My main concern is about the speed here.  Is that equivalent if I use a larger batch size?

Thank you so much in advance! :D

**Comment by MasterSkepticista** [comment]
Fisher Information is taken as expectation over the variance of score *per sample*. PyTorch currently doesn't support batch-computation of gradients *per sample*, hence the batch size of 1. 

You can compute batch gradients *per sample*, by batching forward pass, but disabling loss reduction to mean and `backward()` each loss value of the batch.

Conversely, limit the number of samples used to compute Fisher (say 256), which is good for most use-cases.
